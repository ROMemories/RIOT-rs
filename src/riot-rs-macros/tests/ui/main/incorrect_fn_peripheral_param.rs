#![no_main]
#![feature(type_alias_impl_trait)]
#![feature(used_with_arg)]

// As the macro will fail, this import will not get used
#[allow(unused_imports)]
use riot_rs::embassy::usb::UsbBuilderHook;

// FAIL: the function is expected to take a type having a `take_peripherals()` method as first
// parameter
#[riot_rs::main]
async fn main(_foo: Bar) {}

struct Bar;
