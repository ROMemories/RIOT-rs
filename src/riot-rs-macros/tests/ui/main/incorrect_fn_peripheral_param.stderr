error[E0599]: no method named `take_peripherals` found for mutable reference `&mut OptionalPeripherals` in the current scope
  --> tests/ui/main/incorrect_fn_peripheral_param.rs:11:1
   |
11 | #[riot_rs::main]
   | ^^^^^^^^^^^^^^^^ method not found in `&mut OptionalPeripherals`
   |
   = note: this error originates in the attribute macro `riot_rs::main` (in Nightly builds, run with -Z macro-backtrace for more info)
